x-pack/plugins/observability/common/ui_settings_keys.ts

```diff
@@ -8,3 +8,4 @@
 export const enableInspectEsQueries = 'observability:enableInspectEsQueries';
 export const maxSuggestions = 'observability:maxSuggestions';
 export const enableComparisonByDefault = 'observability:enableComparisonByDefault';
+export const enableInfrastructureView = 'observability:enableInfrastructureView';
```

Comment:
I'm assuming this is an APM specific setting? This file lists Observability-wide settings. We should probably create an APM specific variant.

Reply:
Yes, that sounds right 👍

Reply:
@dgieselaar I checked and the comparison setting key is also specific for APM, so I'm going to create a specific variant for APM and try to export them there.

Reply:
We are going to leave the keys in observability as they are consumed too, and as the [docs](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability/dev_docs/feature_flags.md) suggest